numpy/testing/tests/test_utils.py

```diff
@@ -564,6 +564,16 @@ def test_error_message(self):
         assert_equal(msgs[4], 'Max absolute difference: 2')
         assert_equal(msgs[5], 'Max relative difference: inf')
 
+        # check the error message when x is scalar
+        x = 2
+        y = np.ones(20)
+        with pytest.raises(AssertionError) as exc_info:
+            self._assert_func(x, y)
+        msgs = str(exc_info.value).split('\n')
+        assert_equal(msgs[3], 'Mismatched elements: 20 / 20 (100%)')
+        assert_equal(msgs[4], 'Max absolute difference: 1.')
+        assert_equal(msgs[5], 'Max relative difference: 1.')
+
```

Comment:
Could you split this into its own test? All you would need I think is to add `def test_error_message2(self):` before the lines you added.